### PR TITLE
doc: add custom plugin design doc (#1215)

### PR DESCRIPTION
**Pull request by zen-xu**
ref: #1215 #1218 
Signed-off-by: Xu ZhengYu <zen-xu@outlook.com>

### CODE REVIEW

File: docs/design/custom-plugin.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+func (mp *magicPlugin) OnSessionOpen(ssn *framework.Session) {}
+
+func (mp *magicPlugin) OnSessionClose(ssn *framework.Session) {}
+```
+
+### 2. Build the plugin to .so
+
+vc-scheduler binary is build by `musl-gcc`, so we also need to build the plugin by `musl-gcc`.
```
Comment (k82cn) [inline docs/design/custom-plugin.md:40]: prefer to list known limitations and its options, e.g. different compiler :)

File: docs/design/custom-plugin.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+```yaml
+...
+    containers:
+    - name: volcano-scheduler
+      image: volcano.sh/vc-scheduler:${VERSION}
+      args:
+        - --logtostderr
+        - --scheduler-conf=/volcano.scheduler/volcano-scheduler.conf
+        - -v=3
+        - --plugins-dir=plugins  # specify plugins dir path
```
Comment (Thor-wl) [inline docs/design/custom-plugin.md:94]: So official plugin will be built and loaded as privous way, and custom plugin can be built and loaded in this way.

Comment (zen-xu) [reply docs/design/custom-plugin.md:94]: yes

**Pull request merged**